pkg/util/util.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 VMware, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"encoding/binary"
+	"io"
+)
+
+// Decode decodes data from io reader to specified interfaces
```

Comment:
Maybe an example for interfaces like int or string slice with the predefined size is getting decoded here. As this is being used in multiple places, making it clear is better.